### DISCUSSION TITLE
Part of my code make copilot crash

### DISCUSSION THREAD

**Discussion by Heargo** [community/community#20273] [Copilot Conversations] [answered]
I was using copilot, it was working great and then 2 minutes later there is no suggestions anymore. I had the same issue yesterday but it somehow worked again this morning... Now it's not working again.

I found out that after i wrote certain part of my code copilot doesn't suggest anything anymore.

**I tried :** 
- try the get suggestion on other files/languages
- log out/ log in from the extension
- reload it
- reinstall it
- restart vs code
- restart the computer

Here is the log when I press enter (when there should have a suggestions)

```
[INFO] [default] [2022-07-10T07:59:07.641Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-07-10T07:59:07.737Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 96 ms
```

For a bit of contexte, here is the code
```js
//descriptions is a global object 

export function description(name,age,gender,stats)
{
    var descriptionGenerated="";
    //wealth category
    var familyType="";
    if (stats.wealth>=8) familyType="rich";
    else if(stats.wealth>=6) familyType="aisée";
    else if(stats.wealth>=4) familyType="modeste";
    else familyType="pauvre";

    //baby description 
    if(age<=3)
    {
        //get random baby description
        var descriptionId = Math.floor(Math.random()*descriptions.template.baby[gender].length);
        descriptionGenerated = formated(descriptions.template.baby[gender][descriptionId],{name:name,age:2,face:"test",eyes:"",familyWealth:familyType,future:"nul"});
    }


    //standard description 

    return descriptionGenerated;
    
}
```
and here is the part of code that make copilot crash

```js
//baby description 
if(age<=3)
{
    //get random baby description
    var descriptionId = Math.floor(Math.random()*descriptions.template.baby[gender].length);
    descriptionGenerated = formated(descriptions.template.baby[gender][descriptionId],{name:name,age:2,face:"test",eyes:"",familyWealth:familyType,future:"nul"});
}
```

**Comment by Tostisto** [comment]
Try [this](https://github.com/github-community/community/discussions/10601#discussioncomment-2064057). I could be solution for you.

**Reply by Heargo** [reply]
Thanks, I'll try as soon as I get the problem again (somehow it's not bugged anymore...)

**Comment by DenverCoder1** [comment, marked as answer]
The problem you were facing is probably because the code you wrote uses the word "gender".

It's really frustrating that Copilot has a content filter that it makes it not make any suggestions to code that uses "gender".

I hope they will resolve this at some point.

**Reply by scott88lee** [reply]
As a coding assistant, it should be neutral and adaptive to the user. And not trying to enforce opinionated policy. This is embarrassing.

**Reply by SteshaSoftware** [reply]
hahahaha omfg you CANNOT be serious

**Reply by eatil** [reply]
On the other hand, if I type "fuc" for a variable name, Copilot has no problems with making a suggestion for the rest.

**Reply by ilya-lopukhin** [reply]
lol what's the full stop-words list?
really deep shit, how can you ban words lol

**Reply by pjar** [reply]
This is such an exclusive approach and close-minded. Imagine someone working on a research project about Nazi Germany or LGBT challenges - would just copilot say "Oh come on - you can't do that"?

**Reply by jolitti** [reply]
This is the best publicity for avoiding the copilot. What an intrusive, misguided attempt at preventing discrimination

**Reply by johnmanko** [reply]
🤦

**Reply by Corsico** [reply]
I'm gonna take "white man makes up fictional scenario and gets angry about it" for 300 Jim!

**Reply by bgilner** [reply]
This is the clbuttic mistake of our time. They should support free speech, as our consbreastution does.

**Reply by ghost** [reply]
What's wrong with the word "gender"? 🤔

**Reply by jamesonknutson** [reply]
If you think critically, it’s not the least bit surprising that Copilot refuses to hand out suggestions that could pertain to socially / culturally / politically sensitive topics. Honestly it would be more surprising if it did.

I mean, surely we can all see the potential stories that could come about as a result of copilot generating some suggestion here. Copilot spits out some suggestion that implies that gender is <insert conflicting point of view to loud and opinionated user’s>, and that user goes and posts about it and how copilot is clearly biased towards the <insert opposing group> (think of the examples we have seen where AI models have been proven to be racially biased due to their training set having skewed portions of racial inputs).

This is really just Copilot’s project management avoiding a PR mishap by not letting Copilot even be construed as having an “opinion” on things that could get fleshy humans like us all riled up. Is it annoying? Sure, for a few seconds. Oh well, guess I’ll have to write my boilerplate myself this time around. Is it understandable? Absolutely. GH is just covering their bases and keeping their hands clean of any messes they don’t want to have to clean up, and it makes sense to me as to why this would be one of the areas that Copilot nopes out of providing suggestions.

**Reply by aecra** [reply]
Wouldn't such a strategy foster discriminatory thinking in a non-discriminatory person? If Microsoft don't change their strategy, I won't pay any penny.

**Reply by SandroMartens** [reply]
> 

I'm sure they won't care about your penny.

**Reply by Heargo** [reply]
Thanks, I did few tests and yes the problem comes from that. Do you have the list of the words ?

**Reply by HyeladiBassi** [reply]
@jamesonknutson I understand what you mean here but don't you think it will also cause a problem for the conventional programmers that are using copilot and suddenly have problems because it is not working due to these filters (keep in mind that it is not everyone that knows to check GitHub issues the moment something goes wrong). Personally, I think the issue is a minor issue because you can't blame someone who makes tools because someone else used their tool to commit a crime or act violently against another.

But either way, you are actually right because companies actually care a lot more about the voices of those loud and opinionated users than the users who just want to work in peace. So anyone working on anything even mildly socially, culturally, or politically sensitive will have to work on their own or find ways around when copilot does not provide suggestions.

**Comment by MulleDK19** [comment]
Your code can't contain words like gender, sex, woman, man, male, female, guy, boy, girl, etc.

Yes, it's fucking stupid, but that's how it is..

**Comment by maelp** [comment]
Loooool America is destroying itself because of dumby Trump and dumber Musk
